opentelemetry-api/src/opentelemetry/configuration/__init__.py

```diff
@@ -93,11 +93,13 @@ def __new__(cls) -> "Configuration":
 
             for key, value in environ.items():
 
-                match = fullmatch("OPENTELEMETRY_PYTHON_([A-Z][A-Z_]*)", key)
+                match = fullmatch(
+                    r"OPENTELEMETRY_PYTHON_([A-Za-z_][\w_]*)", key
+                )
 
                 if match is not None:
 
-                    key = match.group(1).lower()
+                    key = match.group(1)
```

Comment:
Why to keep the casing in the python attributes? I think the previous `lower()` was better.

Reply:
Environment variables are case sensitive, so this needs to be too.

Reply:
maybe add a comment as such? I think it might also be good to call out which environments have case sensitive environment variables. I believe windows is case insensitive.